packages/shared/src/components/modals/post/CommentModal.tsx

```diff
@@ -0,0 +1,262 @@
+
+export interface CommentModalProps
+  extends LazyModalCommonProps,
+    CommentMarkdownInputProps {
+  replyToCommentId?: string;
+}
+
+export default function CommentModal({
+  isOpen,
+  onRequestClose,
+  onAfterClose,
+  onCommented,
+  parentCommentId,
+  replyToCommentId,
+  editCommentId,
+  post,
+}: CommentModalProps): ReactElement {
+  const [value, setValue] = useState('');
+  const [styleHeight, setStyleHeight] = useState<number | 'auto'>('auto');
+  const { user } = useAuthContext();
+  const client = useQueryClient();
+  const [modalNode, setModalNode] = useState<HTMLElement>(null);
+
+  const isEdit = !!editCommentId;
+  const isReply = !isEdit && !!replyToCommentId;
+
+  const refCallback = useCallback(
+    (node) => {
+      if (node) {
+        setModalNode(node);
+      }
+    },
+    [setModalNode],
+  );
+
+  const { shouldShowCta, isEnabled, onToggle, onSubmitted } =
+    useNotificationToggle({
+      source: NotificationPromptSource.NewComment,
+    });
+
+  const onSuccess: typeof onCommented = (comment, isNew, parentCommentID) => {
+    if (onCommented) {
+      onCommented(comment, isNew, parentCommentID);
+    }
+    onRequestClose();
+  };
+
+  const comment = useMemo(
+    () =>
+      getCommentFromCache({
+        client,
+        postId: post?.id,
+        commentId: isEdit ? editCommentId : replyToCommentId ?? parentCommentId,
+      }),
+    [
+      client,
+      post?.id,
+      isEdit,
+      editCommentId,
+      parentCommentId,
+      replyToCommentId,
+    ],
+  );
+
+  const { mutateComment, isLoading } = useMutateComment({
+    post,
+    editCommentId: isEdit && editCommentId,
+    parentCommentId: isEdit ? comment?.parent?.id : parentCommentId,
+    onCommented: onSuccess,
+  });
+
+  useLayoutEffect(() => {
+    // scroll to bottom of modal
+    modalNode?.scrollTo({ behavior: 'auto', top: 10000 });
+  }, [modalNode]);
+
+  const { height } = useVisualViewport();
+  const replyHeight = height > 0 ? replySize : 0;
+  const footerHeight = shouldShowCta ? footerSize : 0;
+  const totalHeight = height - headerSize - replyHeight - footerHeight;
+  const inputHeight = totalHeight > 0 ? Math.max(totalHeight, 300) : 'auto';
+
+  useEffect(() => {
+    setStyleHeight(inputHeight);
+  }, [inputHeight, setStyleHeight]);
+
+  const { submitCopy, initialContent } = useMemo(() => {
+    if (isEdit) {
+      return {
+        submitCopy: 'Save',
+        initialContent: comment?.content,
+      };
+    }
+    if (isReply) {
+      return {
+        submitCopy: 'Reply',
+        initialContent:
+          comment?.author?.id && comment.author.id !== user?.id
+            ? `@${comment?.author?.username} `
+            : undefined,
+      };
+    }
+    return { submitCopy: 'Comment' };
+  }, [isEdit, isReply, comment, user?.id]);
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onRequestClose={onRequestClose}
+      onAfterClose={onAfterClose}
+      className="!h-full !max-h-none !w-full !border-none"
```

Comment:
Why all the overwrites?
Don't we just have this type of modal already?

Reply:
Aalso, modals should now be default full-screen on mobile.

Reply:
I think this is just copy & paste from the page, which was not using a modal. Removing 👍